pkg/util/util.go

```diff
@@ -149,6 +153,15 @@ func HaveTolerationsChanged(
 	return !reflect.DeepEqual(cluster.Spec.Placement.Tolerations, existingTolerations)
 }
 
+func DeploymentDeepEqual(d1 *appsv1.Deployment, d2 *appsv1.Deployment) bool {
+	return equality.Semantic.DeepDerivative(d1.Spec.Template.Spec.Containers, d2.Spec.Template.Spec.Containers) &&
```

Comment:
Doesn't this work for the entire deployment spec?

Reply:
Unfortunately no, if a field is empty in the first argument but not empty in the second argument it will return true.